label_studio/data_manager/managers.py

```diff
@@ -192,6 +192,7 @@ def add_result_filter(field_name, _filter, filter_expressions, project):
             Annotation.objects
                 .annotate(json_str=RawSQL('cast(result as text)', ''))
                 .filter(Q(project=project) & Q(json_str__contains=_filter.value))
```

Comment:
Seems that the only reason you are annotating to cast the JSONField into json_str is to run contains on it.  AFAIK Django supports __contains filter lookup on JSONFields (though it is more targeted than the way contains is used here).  Can you clarify why you're doing it like this?  I assume its because we want to match on either keys or values?

Reply:
I didn't modify this line and it seems like it's been this way for a while. Regardless, I think there's a twofold reason why this works best this way.
1. Performance: The requirements for this filter are such that actually parsing the JSON is unnecessary. Just doing a string search is likely faster
2. Variable schema: I think the right side of the  `__contains` argument in a JSONField needs to be an object matching the shape of what's in that JSON field. In annotation.results, the shape of this JSON document and the location of where the actual "label" that we're searching for will vary widely depending on the label config

Reply:
🤦  oops I was digging into the details of the query and completely overlooked that wasn't your change